org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ASTParser.java

```diff
@@ -233,9 +235,24 @@ public static ASTParser newParser(int level) {
 	ASTParser(int level) {
 		DOMASTUtil.checkASTLevel(level);
 		this.apiLevel = level;
+		this.unitResolver = getCompilationUnitResolverInstance();
 		initializeDefaults();
 	}
 
+	@SuppressWarnings("unchecked")
+	private static ICompilationUnitResolver getCompilationUnitResolverInstance() {
```

Comment:
move to ICompilationUnitResolver.getInstance()

Reply:
Good idea, I'll do that.